### PR TITLE
Enable PGPNP and increment wallet version

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

* Enable prefix for phone hashing everywhere
* Enable PGPNP in the wallet
* Increment wallet version
* Implement basic version of exhausted lookup query handling
* Add ability in wallet to wait for tx failure (currently only waits for success)

### Tested

End-to-end In Alfa staging

### Related Issues

Fixes https://github.com/celo-org/celo-labs/issues/465

### Backwards compatibility

No. This invalidates any existing verifications.

Screen Shot 2020-05-13 at 16 27 01

### CODE REVIEW

File: packages/mobile/src/geth/networkConfig.ts [hunk 1]
```diff
@@ -17,50 +17,72 @@ interface NetworkConfig {
...
   blockchainApiUrl: string
+  pgpnpUrl: string // Phone Number Privacy service url
+  pgpnpPubKey: string
 }
 
+const pgpnpUrlStaging = 'https://us-central1-celo-phone-number-privacy-stg.cloudfunctions.net'
+const pgpnpPubKeyStaging =
+  'B+gJTCmTrf9t3X7YQ2F4xekSzd5xg5bdzcJ8NPefby3mScelg5172zl1GgIO9boADEwE67j6M55GwouQwaG5jDZ5tHa2eNtfC7oLIsevuUmzrXVDry9cmsalB0BHX0EA'
+const pgpnpUrlProd = 'https://us-central1-celo-phone-number-privacy.cloudfunctions.net'
+const pgpnpPubKeyProd = 'TODO'
```
Comment (annakaz) [inline packages/mobile/src/geth/networkConfig.ts]: Remove TODO string or leave comment explaining?

**Review by annakaz — APPROVED** [review]
Excited for this to be live!

File: packages/mobile/src/geth/networkConfig.ts (hunk 1 above)
Comment (jmrossy) [reply packages/mobile/src/geth/networkConfig.ts]: I'll be updating this before merging :)

File: packages/mobile/src/geth/networkConfig.ts [hunk 2]
```diff
@@ -17,50 +17,73 @@ interface NetworkConfig {
...
+  pgpnpUrl: string // Phone Number Privacy service url
+  pgpnpPubKey: string
 }
 
+const pgpnpUrlStaging = 'https://us-central1-celo-phone-number-privacy-stg.cloudfunctions.net'
+const pgpnpPubKeyStaging =
+  'B+gJTCmTrf9t3X7YQ2F4xekSzd5xg5bdzcJ8NPefby3mScelg5172zl1GgIO9boADEwE67j6M55GwouQwaG5jDZ5tHa2eNtfC7oLIsevuUmzrXVDry9cmsalB0BHX0EA'
+const pgpnpUrlProd = 'https://us-central1-celo-phone-number-privacy.cloudfunctions.net'
+const pgpnpPubKeyProd =
+  '6VFX/ufxvL54NDRlJMe0jlbb9wD3L/Kfm6K2qEKoxrDMS42Q1S7ZOH88tdpOJhcAJPdDTjGE6qHQBiV48n/jctSuOJ8HjsRzp6VJWMHW3imqep7nwyhzNfLcPXJfrngB'
```
Comment (jeanregisser) [inline packages/mobile/src/geth/networkConfig.ts:29]: These 2 envs don't depend on a specific network?

**Review by jeanregisser — APPROVED** [review]
w00t 👍

File: packages/mobile/src/geth/networkConfig.ts (hunk 2 above)
Comment (jmrossy) [reply packages/mobile/src/geth/networkConfig.ts:29]: For now we're just going to have a prod and staging. Soon we'll set up a slightly different setup for RC1. Then medium-term when the infra architecture is settled, we'll set up one per env :)

File: packages/mobile/src/identity/commentKey.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export function* encryptComment(
   fromAddress: string | null | undefined
 ) {
   // Don't encrypt empty comments
-  if (comment === '') {
+  if (!comment) {
```
Comment (annakaz) [inline packages/mobile/src/identity/commentKey.ts:25]: 🧑‍🎨

File: packages/mobile/src/identity/privacy.ts [hunk 4]
```diff
@@ -24,43 +30,56 @@ export interface PhoneNumberHashDetails {
...
+    e164Number,
+    account,
+    contractKit,
+    selfPhoneHash
+  )
+  yield put(updateE164PhoneNumberSalts({ [e164Number]: details.salt }))
+  return details
+}
+
 // Unlike the getPhoneHash in utils, this leverage the phone number
```
Comment (annakaz) [inline packages/mobile/src/identity/privacy.ts]: Not in the diff but just noticed `leverages` typo

File: packages/mobile/src/identity/privacy.ts [hunk 5]
```diff
@@ -24,43 +30,56 @@ export interface PhoneNumberHashDetails {
...
   } catch (error) {
     if (error.message === ErrorMessages.SALT_QUOTA_EXCEEDED) {
       Logger.error(
         `${TAG}@fetchPrivatePhoneHash`,
         'Salt quota exceeded, navigating to quota purchase screen'
       )
-      // TODO nav to quota purchase screen
+      const result: boolean = yield call(navigateToQuotaPurchaseScreen)
+      // If quota purchase was successful, try lookup a second time
+      if (result) {
```
Comment (annakaz) [inline packages/mobile/src/identity/privacy.ts]: Nit: I would find this more readable if `result` were `quotaPurchaseSucess`

File: packages/mobile/src/identity/privacy.ts [hunk 6]
```diff
@@ -190,3 +209,42 @@ export function* getUserSelfPhoneHash() {
...
+    yield put(
+      transferStableToken({
+        recipientAddress: ownAddress, // send payment to yourself
+        amount: '0.01', // one penny
+        comment: 'Lookup Quota Purchase',
+        txId,
+      })
+    )
+
+    const result = yield call(waitForTransactionWithId, txId)
```
Comment (annakaz) [inline packages/mobile/src/identity/privacy.ts]: Nit: I would find this easier to read if the variable name was more descriptive than `result`

**Review by annakaz — APPROVED** [review]
LGTM! 

A couple nits as suggestions to improve readability

File: packages/mobile/src/identity/privacy.ts (hunk 5 above)
Comment (jmrossy) [reply packages/mobile/src/identity/privacy.ts]: Will do!

File: packages/mobile/src/identity/privacy.ts (hunk 4 above)
Comment (jmrossy) [reply packages/mobile/src/identity/privacy.ts]: Good catch :)

File: packages/mobile/src/identity/privacy.ts (hunk 6 above)
Comment (jmrossy) [reply packages/mobile/src/identity/privacy.ts]: Will do

**Pull request merged**